content-scripts/modules/layout.ts

```diff
@@ -0,0 +1,173 @@
+    if (notificationsList.hasAttribute("data-se-loaded")) return;
+    if (notificationsList.classList.contains("se-loading")) return;
+
+    notificationsList.classList.add("se-loading");
+
+    const newNotifications = document.querySelector<HTMLUListElement>(
+        "#se-auth-new-notifications"
+    );
+    const readNotifications = document.querySelector<HTMLUListElement>(
+        "#se-auth-read-notifications"
+    );
+
+    const dateFormatter = new Intl.DateTimeFormat("pt-PT", {
+        dateStyle: "short",
+    });
+
+    await Promise.allSettled(
+        [
+            [newNotifications, "P"],
+            [readNotifications, "F"],
+        ].map(async (value) => {
+            const [list, type] = value as [HTMLUListElement | null, string];
+            if (!list) return;
+
+            const response = await fetchSigarraPage(
+                `gnots_ajax.show_lista_notifs?pv_estado=${type}`
+            );
+
+            response.querySelectorAll("tr.d").forEach((notification) => {
+                const date = notification.querySelector("td:nth-child(3)")?.textContent;
+                const title = notification.querySelector("td:nth-child(4)")?.textContent;
+                const answer = notification.querySelector<HTMLInputElement>(
+                    "td:nth-child(7) input"
+                );
+
+                //TODO(thePeras): Could be a jsx component
+                const li = document.createElement("li");
+                li.classList.add("se-notification");
+
+                const time = document.createElement("time");
+                time.classList.add("se-notification-time");
+                if (date) {
+                    time.dateTime = date;
+                    time.textContent = dateFormatter.format(new Date(date));
+                }
+
+                const span = document.createElement("span");
+                span.classList.add("se-notification-title");
+                if (title) {
+                    span.textContent = title;
+                }
+
+                li.append(span, time);
+
+                if (answer) {
+                    const markAsRead = async (e: Event) => {
+                        e.preventDefault();
+                        e.stopPropagation();
+
+                        await fetchSigarraPage(
+                            `gnots_geral.nots_list_sub?${answer.name}=${answer.value}`
+                        );
+
+                        const targetElement = e.target as HTMLElement;
+                        targetElement.remove();
+
+                        readNotifications?.insertBefore(
+                            li,
+                            readNotifications.firstChild
+                        );
+                    };
+
+                    li.append(Button({
+                        icon: "ri-check-line",
+                        className: "se-notification-button",
+                        onClick: markAsRead,
+                    }));
+                }
```

Comment:
There seems to be a bug here :P
![image](https://github.com/user-attachments/assets/16b77afd-c0e3-4f5b-9b08-c258cc34ea93)

Reply:
Should be fixed! Check

Reply:
I was still getting the same error, but this seemed to fix it:
` {title !== undefined ? (<span>{title}</span>) : ""}`

Reply:
`onClick` should actually be `onclick` :P

Reply:
done!